screens/GameDetails.js

```diff
@@ -0,0 +1,301 @@
+import React, { useEffect, useState } from "react";
+import { Auth } from "aws-amplify";
+import EStyleSheet from "react-native-extended-stylesheet";
+import { View, Text, TouchableOpacity, FlatList } from "react-native";
+import Button from "../common/Button";
+import BackArrow from "../common/BackArrow";
+import { DataStore } from "aws-amplify";
+import "@azure/core-asynciterator-polyfill";
+import {
+  Player,
+  Game,
+  Location,
+  GamePlayer,
+  Rsvp,
+  SkillLevel,
+} from "../src/models";
+
+import {epochToLocalDate} from '../utils/TimeUtil';
+import {epochToLocalTime} from '../utils/TimeUtil';
+import rsvp from "../utils/rsvp";
+
+function GameDetails({ route, navigation }) {
+  const details = route.params.item;
+  console.log("detuals", details)
+  const [loading, setLoading] = useState(false);
+  const [accepted, setAccepted] = useState([]);
+  const [declined, setDeclined] = useState([]);
+  const [thisPlayer, setThisPlayer] = useState([]);
+
+  const [gameOrganizer, setGameOrganizer] = useState([]);
+
+  //gets current user
+  async function getPlayer() {
+    setLoading(true);
+    try {
+      const response = await Auth.currentUserInfo();
+      const player = await DataStore.query(Player, (p) => p.email.eq(response.attributes.email));
+      console.log("This player returned: ", player[0]);
+      // setThisPlayer(player[0]);
+      return player[0];
+
+
+    } catch (error) {
+      console.log("Error getting player: ", error.message);
```

Comment:
Nothing is done to alert the calling function that an error occurred. Make sure to have a null check or something in the useEffect for when this function fails. Also, make sure you setLoading(false) or else you have an infinite loading screen.